pyhealth/datasets/base_dataset.py

```diff
@@ -76,7 +76,6 @@ class BaseDataset(ABC, Dataset):
 
     def __init__(
             self,
-            dataset_name: str,
```

Comment:
I think we should still keep this arg for flexibility. It this arg is not provided, we will use self.__class__.__name__

Reply:
E.g., `dataset_name: Optional[str] = None`